source/main_module/cooling/cooling.c

```diff
@@ -32,13 +35,50 @@ bool coolingInit()
     // NVIC_EnableIRQ(EXTI9_5_IRQn);
 
 
+    //Init Timer
+    int arrValue = 10;
+
+    TIM4 -> CR1 &= ~TIM_CR1_CEN; //Turning off counter
+
+    RCC -> APB1ENR1 |= RCC_APB1ENR1_TIM4EN; //0b100;
+    TIM4 -> PSC = (APB1ClockRateHz / (PWM_FREQUENCY * arrValue)) - 1; 
+    TIM4 -> ARR = arrValue - 1; //setting it to 99 so it's easier to use it with Duty Cycle
+
+    //Set Channels 1 and 2 to 110 (Mode 1 up counter) -> (active while CNT <= CCR)
+    TIM4 -> CCMR1 |= TIM_CCMR1_OC1M_2 | TIM_CCMR1_OC1M_1; 
+    TIM4 -> CCMR1 |= TIM_CCMR1_OC2M_2 | TIM_CCMR1_OC2M_1;
+
+    //Enable Channels 1 and 2 outputs 
+    TIM4 -> CCER |= TIM_CCER_CC1E | TIM_CCER_CC2E;
+
+    //Setting pwm to 0
+    TIM4 -> CCR1 = 0;
+    TIM4 -> CCR2 = 0;
+
+    //Enable counter as long as ccrs are 0
+    if ((TIM4 -> CCR1 == 0) && (TIM4 -> CCR2 == 0))
```

Comment:
I meant the CCR1 and CCR2 could be non-zero before the init function is called. Once you set then to 0, you don’t have to check if they are 0 (no need for if statement)